plugins/modem-manager/fu-mbim-qdu-updater.c

```diff
@@ -0,0 +1,545 @@
+{
+	CloseContext *ctx = (CloseContext *) user_data;
+
+	/* ignore errors when closing */
+	mbim_device_close_finish (MBIM_DEVICE (mbim_device), res, (ctx->error == NULL) ? &ctx->error : NULL);
+	g_clear_object (&ctx->mbim_device);
+	g_main_loop_quit (ctx->mainloop);
+}
+
+static gboolean
+fu_mbim_qdu_updater_reprobe_timeout (gpointer user_data)
+{
+	CloseContext *ctx = user_data;
+
+	ctx->timeout_id = 0;
+	g_main_loop_quit (ctx->mainloop);
+
+	return G_SOURCE_REMOVE;
+}
+
+gboolean
+fu_mbim_qdu_updater_close (FuMbimQduUpdater *self, GError **error)
+{
+	g_autoptr(GMainLoop) mainloop = g_main_loop_new (NULL, FALSE);
+	g_autofree gchar *device_sysfs_path = NULL;
+	CloseContext ctx = {
+		.mainloop = mainloop,
+		.mbim_device = g_steal_pointer (&self->mbim_device),
+		.timeout_id = 0,
+	};
+
+	mbim_device_close (ctx.mbim_device, 5, NULL, fu_mbim_qdu_updater_mbim_device_close_ready, &ctx);
+	g_main_loop_run (mainloop);
+
+	/* we should always have both device cleared, and optionally error set */
+
+	g_warn_if_fail (ctx.mbim_device == NULL);
+
+	if (ctx.error != NULL) {
+		g_propagate_error (error, ctx.error);
+		return FALSE;
+	}
+ 
+	fu_mm_utils_get_port_info (self->mbim_port, &device_sysfs_path, NULL, NULL);
+	/* device will auto reboot after update */
+	/* pcie rescan is required */
+	/* mbim-proxy stuck due to mbim port suddenly lost */
+	if (device_sysfs_path) {
+		g_auto(GStrv) split = NULL;
+		split = g_strsplit (device_sysfs_path, "/", -1);
+		/* /sys/devices/pci0000:00/0000:00:1d.3/0000:02:00.0 */
+		if (g_strv_length (split) == 6) {
+			const gchar *remove_path = g_strdup_printf ("/sys/bus/pci/devices/%s/%s/remove", split[4], split[5]);
+			const gchar *rescan_path = "/sys/bus/pci/rescan";
+			gchar value;
+			FILE *f;
+			g_auto(GStrv) argc = NULL;
+			GError *tmp_error = NULL;
+			gint exit_stat = 0;
+
+			if (!(f = fopen (remove_path, "w"))) {
+				g_debug ("fail to open %s", remove_path);
+			}
+
+			if (f) {
+				value = '1';
+				if (fwrite (&value, 1, 1, f) != 1) {
+					g_debug ("fail to write %s", remove_path);
+				}
+			}
+
+			if (f)
+				fclose(f);
+
+			argc = g_new0 (gchar *, 4);
+			argc[3] = NULL;
+			argc[0] = g_strdup ("killall");
+			argc[1] = g_strdup ("-9");
+			argc[2] = g_strdup ("mbim-proxy");
+			if (!g_spawn_sync (NULL,
+					  argc,
+					  NULL,
+					  G_SPAWN_SEARCH_PATH | G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL,
+					  NULL,
+					  NULL,
+					  NULL,
+					  NULL,
+					  &exit_stat,
+					  &tmp_error)) {
+				g_debug ("error spawning kill mbim-proxy: %s ", tmp_error->message);
+				g_clear_error (&tmp_error);
+			}
+
+			if (!(f = fopen (rescan_path, "w"))) {
+				g_debug ("fail to open %s", rescan_path);
+			}
+
+			if (f) {
+				value = '1';
+				if (fwrite (&value, 1, 1, f) != 1) {
+					g_debug ("fail to write %s", rescan_path);
+				}
+			}
+
+			if (f)
+				fclose (f);
+			g_debug ("%s", remove_path);
```

Comment:
These logs don't seem useful at all, just printing paths without even saying what each path is. Looks to me more a debugging trace used during development than a proper debug log